### ISSUE TITLE
Add support for Oracle

### ISSUE THREAD

**Issue by pierre** [killbill/killbill#1036]
This is the master ticket to track the work involved to support Oracle:

- A `ddl-oracle.sql` (similar to `ddl-postgresql.sql`) needs to be created
- An Oracle specific [locker](https://github.com/killbill/killbill-commons/tree/master/locker/src/main/java/org/killbill/commons/locker) needs to be implemented, using `GET_LOCK` and `RELEASE_LOCK`
- Tests must support a `localtest-oracle` profile for regression testing
- In Kill Bill, `GlobalLockerModule` will need to be updated
- Update the documentation on how to deploy the Oracle JDBC jar

Optional work to be considered:
- Add Oracle support for Ruby plugins and generate the DDL file from ActiveRecord migrations in plugins (such as [Stripe](https://github.com/killbill/killbill-stripe-plugin/blob/master/db/ddl.sql))
- Update the [Analytics plugin](https://github.com/killbill/killbill-analytics-plugin) DDL and queries

Similar to https://github.com/killbill/killbill/issues/273.

**Comment by dowonderatwill** [comment]
Wanted to know whether ddl-oracle.sql has progressed? I was trying to migrate the ddl to oracle but one hurdle is that names of some of the table is more than 30 characters, e.g;catalog_override_phase_definition  and Oracle 11g does not support more than 30 chars in the table name! What could be the workaround for it?

**Comment by pierre** [comment]
I had a private fork with initial support for Oracle Database 19c (19.3.0) and Oracle WebLogic Server 12c R2 (12.2.1.3). It was _mostly_ working, but there were quite a few (non trivial) TODOs left to finish the work:

* Support for export endpoint
* Support for pagination/search endpoints
* Support for Shiro sessions
* GlobalLocker implementation

Due to the lack of interest by the community (over the years, we've only had a few companies interested in Oracle support and most of them have happily adopted PostgreSQL instead), this is currently not on our radar.

If this is a blocker for you, feel free to reach out to me directly (`pierre @ <domain>`) to discuss your options.

**Comment by dowonderatwill** [comment]
Eventually I also moved to MySql and time being this is not blocker. But just wanted to know how did you manage the table name exceeding 30 characters? Changed the table name to be smaller than that and then changed in source code where the table names are used?

**Comment by pierre** [comment]
Yup, I had DDL and code modifications (e.g. how we handle auto generated keys).
